src/components/Firestore/Requests/RequestPath/index.test.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { act, fireEvent, render, waitFor } from '@testing-library/react';
+import React from 'react';
+
+import RequestPath from './index';
+
+describe('RequestPath', () => {
+  const MOCKED_PATH =
+    '/collection1/collection1_ID/subcollection/subcollection_ID';
+  const COPY_ICON_TEXT = 'content_copy';
+
+  describe('RequestPath => truncation', () => {
+    // Manually set a custom offsetWidth value of a given HTML element
+    // NOTE: using this mocked values is important because jsdom does not handle
+    // layouts (which means that every variable that contains a size measurement
+    // will be equal to 0 during a test)
+    function setHTMLElementOffsetWidth(
```

Comment:
Move this function to the top-level of the file since it doesn't access any of the enclosing state